datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaProducerWrapper.java

```diff
@@ -390,6 +398,34 @@ void flush() {
     if (producer != null) {
       try {
         producer.flush(_producerFlushTimeoutMs, TimeUnit.MILLISECONDS);
+      } catch (IllegalStateException e) {
+        _log.warn("Hitting IllegalStateException during kafka producer flush. Wait for the producer to close.", e);
```

Comment:
Move this log to where you check for close in progress?
Add a different warn log if it isn't close in progress?

Reply:
done.